bananas_api/new_upload/readers/heightmap.py

```diff
@@ -39,6 +47,50 @@ def read(self, fp):
 
         try:
             im = Image.open(fp)
-            self.size = im.size
         except Exception:
             raise ValidationException("File is not a valid image.")
+
+        # Limit the size of heightmaps, as otherwise it could allocate a lot of
+        # memory for clients loading the map.
+        if im.width * im.height > 16384 * 16384:
```

Comment:
Similar: reject if "width < 256 or height < 256". Such small heightmaps are not really useable.

Reply:
Went with `Resolution.LOW` for these maps, as we already have a few that are smaller.